org.eclipse.acute/src/org/eclipse/acute/dotnetnew/DotnetNewWizard.java

```diff
@@ -101,10 +131,83 @@ public boolean performFinish() {
 				}
 			}
 		} catch (CoreException e) {
-			MessageDialog.openError(getShell(), "Unable to load project description", "");
+			MessageDialog.openError(getShell(), "Error", "Unable to load project description.");
 			return false;
 		}
 		return true;
 	}
 
+	private Set<IWorkingSet> getWorkingSets(Object o) {
+		IWorkingSet[] allWorkingSets = PlatformUI.getWorkbench().getWorkingSetManager().getAllWorkingSets();
+		Set<IWorkingSet> fileWorkingSets = new HashSet<>();
+
+		for (IWorkingSet iWorkingSet : allWorkingSets) {
+			IAdaptable[] elements = iWorkingSet.getElements();
+
+			for (IAdaptable element : elements) {
+				if (element.equals(o)) {
+					fileWorkingSets.add(iWorkingSet);
+					break;
+				} else if (element instanceof IContainer) {
+					IContainer container = (IContainer) element;
+					boolean objectFound = false;
+
+					for (IResource resource : recursiveResources(container)) {
+						if (resource.equals(o)) {
+							fileWorkingSets.add(iWorkingSet);
+							objectFound = true;
+							break;
+						}
+					}
+					if (objectFound) {
+						break;
+					}
+
+				}
+			}
+		}
+
+		return fileWorkingSets;
+	}
+
+
+	private Set<IResource> recursiveResources(IContainer container) {
+		Set<IResource> resources = new HashSet<>();
+		try {
+			for (IResource resource : container.members()) {
+				if (resource instanceof IContainer) {
+					resources.addAll(recursiveResources((IContainer) resource));
+				} else if (((IAdaptable) resource).getAdapter(IContainer.class) != null) {
+					IContainer subFolder = ((IAdaptable) resource).getAdapter(IContainer.class);
+					resources.addAll(recursiveResources(subFolder));
+				} else {
+					resources.add(resource);
+				}
+			}
+		} catch (CoreException e) {
+			return Collections.emptySet();
+		}
+		return resources;
+	}
+
+	private File toFile(Object o) {
+		if (o instanceof File) {
+			return (File) o;
+		} else if (o instanceof IResource) {
+			IPath location = ((IResource) o).getLocation();
+			if(location.toFile().isFile()) {
+				return location.toFile().getParentFile().getAbsoluteFile();
+			}
+			return location == null ? null : location.toFile();
+		} else if (o instanceof IAdaptable) {
+			IResource resource = ((IAdaptable) o).getAdapter(IResource.class);
```

Comment:
You might even be able to try `((IAdaptable)o).getAdapter(File.class)` in case this one returns null.